lib/service_now_request.rb

```diff
@@ -20,13 +23,15 @@ def print_response
                     use_ssl: @uri.scheme == 'https',
                     verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
       header = { 'Content-Type' => 'application/json' }
+      # Add bearer token to header if using oauth
+      header['Authorization'] = "Bearer #{@access_token}" if @access_token
```

Comment:
I think credentials code should be in one place. So, line 34 should move up here, making this look something like
```
if @access_token
  // Token auth
else
  // Username/password auth
end
```

Reply:
These are separated so that the http gem can encode the headers properly when creating the request object. Once the request object is created we can use the basic_auth method which also handles the encoding. I tried to have them both be headers, but kept running into encoding issues which I think the basic_auth method is handling for us.

Reply:
Ah yeah, I didn't see the request object being created.

Looks like you should be able to do `request[<header>] = <value>` so I think if you move the `Authorization` header code down below, you should be able to have the credentials in one place. So something like

```
if @access_token
  request['Authorization'] = ...
else
  # username/password
  request.basic_auth(...)
end
```

Reply:
oh nice, updated.